docs/guide/docs/infrastructure/development-pipeline.md

```diff
@@ -0,0 +1,115 @@
+---
+id: development-lifecycle
+title: Development Lifecycle
+---
+
+## Git Syncing – Source Control Management
+
+A best practice is to keep track of changes to your chatbot using your preferred [Source Control Management Tool(SCM)](https://www.softwaretestinghelp.com/version-control-software/) and always deploy the master branch in production. Once deployed, you can regularly [pull](versions#pull) changes and update them to your SCM or revert to them when the need arises. Doing so helps you harness your SCM power for branches, merging conflicting files, reviewing changes, and creating revisions.
+
+Suppose you have a more complex deployment pipeline with one or multiple staging environments with pending changes on each environment. In that case, you can easily track and work on them using development pipelines.
+
+### Setting up Botpress Git Syncing (with GitHub)
+
+1. [Download Botpress](https://botpress.com/download)
+2. Open your terminal and ```cd [PATH]/botpress-v12_20_2-darwin-x64```
+3. Set up GitHub remote tracking
+```
+git init
+git add README.md
+git commit -m "first commit"
+git branch -M main
+git remote add origin git@github.com:[GH_USERNAME]/bp-project.git
+git push -u origin main
+```
+4. You can add a develop and staging branch to fit your workflow
+5. Everytime you make a change in the Conversation Studio or in your preferred code editor, you can commit that change to track changes.
+6. When you are ready to send you chatbot to production, you can ```git merge``` your develop branch in master.
+7. Then, you ```checkout master``` in your terminal and you go to your Botpress Admin to export your chatbot.
+8. You upload it on your production environment (usually cloud deployment)
+
+### Syncing Changes Between Environments
+
+We will use Git to sync changes between 2 environments and promote an environment (i.e., promote staging to production).
+
+Let's assume that given a pipeline with three environments, **development**, **staging**, and **production**, there are some changes both on production and staging, and you want to promote staging to production. What we want to do is the following:
+
+1. Create a merge conflict so we can choose what we want in a merge conflict tool.
+2. Resolve conflicts (i.e., merge staging into production)
```

Comment:
I think it would make more sense if you said something like: 

```suggestion
1. Merge production into staging which might result in a merge conflict.
2. Resolve conflicts using any merge conflict tool (e.g., VSCode, Sublime Merge)
```

Personally, it's the "Create a merge conflict" part that stood out. I find it weird to say that you want to create a merge conflict since a merge conflict is a side effect of merging two branches and not something you really intend to do.